x-pack/plugins/security_solution/public/management/components/console/service/parsed_command_input.ts

```diff
@@ -157,3 +157,15 @@ export const getArgumentsForCommand = (command: CommandDefinition): string => {
 
   return `${requiredArgs} ${optionalArgs.length > 0 ? `[${optionalArgs}]` : ''}`.trim();
 };
+
+export const parsedPidOrEntityIdParameter = (
+  pid: string | number
+): { pid: number; entity_id?: never } | { pid?: never; entity_id: string } => {
+  const parsedPid = !isNaN(Number(pid)) ? Number(pid) : pid;
```

Comment:
Inputs with perhaps undesired outcomes:

- `'37.37'` will be parsed as a number, and will give us `{ pid: 37.37 }`. Probably not a valid PID. Might be an entity if there's a dot, but I missed the discussion on what valid entity ids are. 
- `''` that is, the empty string. I think this gives the output `{ pid: 0 }` unless it is filtered out elsewhere for being falsy
- `'    '` a string of just spaces. Will not get filtered out by falsy tests, and will also be type-coerced to the Number 0

Reply:
[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#useful_special-case_behavior) also says `isNaN(Number(x))` is strictly equivalent to `isNaN(x)`

Reply:
all good points, will refactor logic and add more tests

Reply:
@pzl this logic has also been removed/refactored since we are back to accepting one of two different parameters